f-account-info@v0.11.0 - Wired up the store action `editConsumerDetails` to the Template

oliversweb opened the pull request:
### Changed
- Wired up the Vuex Store `editConsumerDetails` action to the Template

Commit:
Wired up the store action `editConsumerDetails` to the Template

kevinrodrigues commented on packages/components/pages/f-account-info/src/components/AccountInfo.vue:
```diff
@@ -310,15 +320,12 @@ export default {
-        editConsumerDetails (field, value) {
+        * A generic method that updates the State (e.g. 'consumer.<field> = value')
+        * @param {string} field - The field of the consumer that needs changing
+        * @param {string} value - The new value the consumer field
```
`The new value of the consumer field`?

kevinrodrigues commented on packages/components/pages/f-account-info/src/components/AccountInfo.vue:
```diff
@@ -310,15 +320,12 @@ export default {
+        * @param {string} field - The field of the consumer that needs changing
+        * @param {string} value - The new value the consumer field
+        */
+        editConsumer (field, value) {
```
Maybe we can call this `onEditConsumerDetails` or `onEditConsumer`?

kevinrodrigues commented on packages/components/pages/f-account-info/src/components/_tests/AccountInfo.test.js:
```diff
@@ -154,20 +154,39 @@ describe('AccountInfo', () => {
         });
     });
 
-    describe('`methods`', () => {
```
Personally i think this describe makes it easier to keep things (method tests in this case) together, otherwise it gets hard to find places to add new tests.

Commit:
Review tweaks

Commit:
After wiring into CoreWeb needed to fix up some issues.
Api spec wrong and get addresses returns an Addresses object with an array of addresses.
The form fields were wired up using v-model but vuex store state is not bidirectional.
The register store module needed to be moved to beforeCreate() because the store action helpers are wired up before 'created()' and after 'beforeCreate()'.
Also fixed up tests/data.
